src/app/shared/components/multi-select/multi-select.component.ts

```diff
@@ -49,6 +55,29 @@ export class MultiSelectComponent<T> implements OnInit, ControlValueAccessor {
         isSelected: false,
       }
     })
+    this.clickService.clickObservable.pipe(takeUntil(this.destroySubject)).subscribe(click => this.onDocumentClick(click))
+  }
+
+  private onDocumentClick(clickEvent: MouseEvent): void {
+    if (!this.isShown) {
+      return
+    }
+    if (!this.isMultiSelectElementClicked(clickEvent)) {
+      this.isShown = false
+    }
+  }
+
+  private isMultiSelectElementClicked(clickEvent: MouseEvent): boolean {
+    const allClassNames = []
+    let clickTarget: Element = clickEvent.target as Element
+    while (clickTarget) {
+      allClassNames.push(...clickTarget.classList.value.split(' '))
+      if (!clickTarget.parentElement) {
+        break
+      }
+      clickTarget = clickTarget.parentElement
+    }
+    return !!allClassNames.find(className => className === this.className)
```

Comment:
Having the click service seems overly complex and makes the multi select depend on external components to work. I am not sure what the classnames are for.

I suggest an alternative solution, where when opening the options overlay, adds an event listener to the document, which is removed once it is triggered (with the `once` option). Then the multi-select component just needs to stop the event bubbling. I have tried implementing it in the TypeScript file, these things should be added (based on `develop`):
```typescript
export class MultiSelectComponent implements ... {
  @HostListener('click', ['$event'])
  protected stopEventPropagation(event: MouseEvent): void {
    event.stopPropagation()
  }

  protected isShowingOptions: boolean = false

  protected toggleIsShowingOptions(): void {
    this.isShowingOptions = !this.isShowingOptions
    if (this.isShowingOptions) {
      document.addEventListener('click', () => (this.isShowingOptions = false), { once: true })
    }
  }
}
```

In the template, the `div.display-area` gets `(click)="toggleIsShowingOptions()"` to toggle the visibility of the options overlay and `div.multi-select-dropdown` gets `*ngIf="isShowingOptions"` to react to the toggling. And the hover selectors in the css should be removed.

Reply:
My approach was with a  ClickService to avoid each component creating its own `HostListener`. I made the desired change, but you should keep in mind going forward that if you have a bunch of components on one page and each one has a `HostListener`, it will slow down the overall performance of the page.

Reply:
It should only be an issue if we have multiple components registered at the same time.
But since each component should unregister itself after it closes (and each component should close itself whenever another opens), we should never have more than one listening at a time.

Reply:
Valid points :) Though it would require a quite hefty amount of listeners for it to become an issue.
As @LindvedKrvang mentions in a later comment – I did not take the scenario of the interaction between multiple selects into account.